contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,234 @@
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+
+        uint256 shares = _underlyingToShares(_amount);
+
+        _resetPendingWithdrawalIfEpochAdvanced();
+
+        // because the vault can call updateInvested => withdrawToVault multiple times during the same epoch,
+        // and rysk liquidity pool aggregates withdrawal amounts inititated during the same epoch,
+        // we cannot withdraw less than already pending amount
+        if (pendingWithdrawal.shares >= shares) return;
+
+        uint256 sharesToWithdraw = shares - pendingWithdrawal.shares;
+        if (!_hasEnoughShares(sharesToWithdraw))
+            revert StrategyNotEnoughShares();
+
+        pendingWithdrawal.shares += uint128(sharesToWithdraw);
+
+        emit RyskWithdrawalInitiated(sharesToWithdraw);
+        ryskLqPool.initiateWithdraw(sharesToWithdraw);
+    }
+
+    /**
+     * Completes the withdrawal initiated by withdrawToVault.
+     *
+     * @notice Expected to be called by the backend once the Rysk liquidity pool enters a new epoch.
+     * Backend should subscribe to 'EpochExecuted' event on Rysk LiquidityPool contract,
+     * and act by calling this function when the event is emitted to complete ongoing withdrawal.
+     */
+    function completeWithdrawal() external {
+        if (pendingWithdrawal.epoch == 0) revert RyskNoWithdrawalInitiated();
+        if (pendingWithdrawal.epoch == ryskLqPool.epoch())
+            revert RyskCannotComipleteWithdrawalInSameEpoch();
+
+        uint256 sharesToWithdraw = ryskLqPool
+            .withdrawalReceipts(address(this))
+            .shares;
+
+        uint256 amountWithdrawn = ryskLqPool.completeWithdraw(sharesToWithdraw);
+
+        emit StrategyWithdrawn(amountWithdrawn);
+        underlying.safeTransfer(vault, amountWithdrawn);
+    }
+
+    /**
+     * Get the number of Rysk liquidity pool shares owned by the strategy.
+     *
+     * @return shares owned by the strategy
+     */
+    function _getTotalShares() internal view returns (uint256) {
+        // total shares = redeemed shares + unredeemed shares
+        return
+            ryskLqPool.balanceOf(address(this)) +
+            ryskLqPool.depositReceipts(address(this)).unredeemedShares;
+    }
+
+    /**
+     * Calculates the value of Rysk liquidity pool vault shares in underlying.
+     *
+     * @param _shares number of shares
+     *
+     * @return underlying value of shares
+     */
+    function _sharesToUnderlying(uint256 _shares)
+        internal
+        view
+        returns (uint256)
+    {
+        return
+            (_shares * ryskLqPool.epochPricePerShare(ryskLqPool.epoch())) /
```

Comment:
would need to account for the new epoch change. also this may lead to stale update in the share price